src/api/join_element.rs

```diff
@@ -302,8 +248,28 @@ impl<Packet: Sized> Stream for JoinElementProvider<Packet> {
             }
         }
 
-        let task = task::current();
-        if self.await_consumers.try_send(task).is_err() {
+        // For now, we are going to store our task handle in every task_park, since we do not have a good way to let
+        // the cosumers share one big task park. Perhaps we could use a mutexed queue, or something similar, rather than
+        // spraying the task handle to every consumer. I think the current implementation results in the provider being
+        // scheduled quite frequently.
+        let provider = task::current();
```

Comment:
`provider_task`?

Reply:
I left it as provider and consumer rather than provider_task and consumer_task, cause provider.notify reads better.